### PR TITLE
Switch tests to the Rust Frontegg Mock

### PR DESCRIPTION
**Pull request by bobbyiliev**
As per the #461 epic, now that we have all of the required endpoints transferred over, removing the Frontegg Mock Golang implementation in favour of the Rust Frontegg mock service from the main Materialize repo.

### CODE REVIEW

File: pkg/provider/acceptance_scim_config_test.go [hunk 1]
```diff
@@ -27,7 +27,8 @@ func TestAccSCIM2Configuration_basic(t *testing.T) {
 					testAccCheckSCIM2ConfigurationExists("materialize_scim_config.scim_config_example", source, connectionName),
 					resource.TestCheckResourceAttr("materialize_scim_config.scim_config_example", "source", source),
 					resource.TestCheckResourceAttr("materialize_scim_config.scim_config_example", "connection_name", connectionName),
-					resource.TestCheckResourceAttrSet("materialize_scim_config.scim_config_example", "tenant_id"),
+					// TODO: Mock services need to be updated to return these values
+					// resource.TestCheckResourceAttrSet("materialize_scim_config.scim_config_example", "tenant_id"),
```
[the remark this answers is not in the document]
Comment (bobbyiliev) [reply pkg/provider/acceptance_scim_config_test.go:31]: PR with a fix for this: https://github.com/MaterializeInc/materialize/pull/29149

File: pkg/provider/acceptance_user_test.go [hunk 2]
```diff
@@ -27,11 +27,11 @@ func TestAccUser_basic(t *testing.T) {
 					resource.TestCheckResourceAttr("materialize_user.example_user", "email", email),
 					resource.TestCheckResourceAttr("materialize_user.example_user", "send_activation_email", "false"),
 					resource.TestCheckResourceAttr("materialize_user.example_user", "roles.0", "Member"),
-					resource.TestCheckResourceAttr("materialize_user.example_user", "verified", "false"),
+					resource.TestCheckResourceAttr("materialize_user.example_user", "verified", "true"),
 					// Data source tests
 					resource.TestCheckResourceAttrPair("data.materialize_user.user_data", "id", "materialize_user.example_user", "id"),
 					resource.TestCheckResourceAttr("data.materialize_user.user_data", "email", email),
-					resource.TestCheckResourceAttr("data.materialize_user.user_data", "verified", "false"),
+					resource.TestCheckResourceAttr("data.materialize_user.user_data", "verified", "true"),
```
Comment (RobinClowers) [inline pkg/provider/acceptance_user_test.go:34]: Is this just a change in the mock behavior?

**Review by RobinClowers — APPROVED** [review]
Awesome to see this land, I'm sure it was a ton of work!

File: pkg/provider/acceptance_user_test.go (hunk 2 above)
Comment (bobbyiliev) [reply pkg/provider/acceptance_user_test.go:34]: Yep, we decided to default to `true`:

https://github.com/MaterializeInc/materialize/blob/5ea2a212000a9d28dbe758879aa23c7fcfe44a20/src/frontegg-mock/src/lib.rs#L754

**Pull request merged**